go/mcap/indexed_message_iterator.go

```diff
@@ -144,9 +165,48 @@ func (it *indexedMessageIterator) parseSummarySection() error {
 			it.statistics = stats
 		case TokenFooter:
 			it.hasReadSummarySection = true
-			return nil
+			break lexerloop
+		}
+	}
+	it.chunksOverlap = false
+	switch it.indexHeap.order {
+	case FileOrder:
+		sort.Slice(it.chunkIndexesToLoad, func(i, j int) bool {
+			return it.chunkIndexesToLoad[i].ChunkStartOffset < it.chunkIndexesToLoad[j].ChunkStartOffset
+		})
+	case LogTimeOrder:
+		sort.Slice(it.chunkIndexesToLoad, func(i, j int) bool {
+			return it.chunkIndexesToLoad[i].MessageStartTime < it.chunkIndexesToLoad[j].MessageStartTime
+		})
+		for i := 1; i < len(it.chunkIndexesToLoad); i++ {
+			prev := it.chunkIndexesToLoad[i-1]
+			cur := it.chunkIndexesToLoad[i]
+			if prev.MessageEndTime > cur.MessageStartTime {
+				it.chunksOverlap = true
+				break
+			}
+		}
+	case ReverseLogTimeOrder:
+		sort.Slice(it.chunkIndexesToLoad, func(i, j int) bool {
+			return it.chunkIndexesToLoad[i].MessageEndTime > it.chunkIndexesToLoad[j].MessageEndTime
+		})
+		for i := 1; i < len(it.chunkIndexesToLoad); i++ {
+			prev := it.chunkIndexesToLoad[i-1]
+			cur := it.chunkIndexesToLoad[i]
+			if prev.MessageStartTime < cur.MessageEndTime {
+				it.chunksOverlap = true
+				break
+			}
 		}
 	}
+	if it.chunksOverlap {
```

Comment:
I think we probably need to expand the commenting of this function, both in the docstring and inline, to describe the approach it is implementing.